### PR TITLE
Initial hardware code

### PR DESCRIPTION
**Pull request by jamiesanson**
# ( ͡° ͜ʖ ͡°)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# base-station
+
+A base station to run a rocket, and display some pretty graphs
+
+## How to run
+### Hardware code
+* Construct circuit shown in schematic
+* Move folders in ```hardware/src/libs``` to global Arduino library directory (usually ```Documents/Arduino/libraries```)
```
Comment (marcelvanworkum) [inline README.md:8]: Could create an `init.sh` to do this, but probably way overkill

Comment (jamiesanson) [reply README.md:8]: Probably a good idea, but something to do on the side I guess as it's not really related to any scope

File: hardware/src/libs/ESP8266/ESP8266.cpp [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+    start = millis();
+    while (millis() - start < timeout) {
+        if(m_puart->available() > 0) {
+            a = m_puart->read();
+            data += a;
+        }
+
+        index_PIPDcomma = data.indexOf("+IPD,");
+        if (index_PIPDcomma != -1) {
+            index_colon = data.indexOf(':', index_PIPDcomma + 5);
```
Comment (marcelvanworkum) [inline hardware/src/libs/ESP8266/ESP8266.cpp]: magic numbers?

File: hardware/src/libs/ESP8266/ESP8266.cpp [hunk 3]
```diff
@@ -0,0 +1,721 @@
...
+                break;
+            }
+        }
+    }
+
+    if (has_data) {
+        i = 0;
+        ret = len > (int32_t) buffer_size ? buffer_size : len;
+        start = millis();
+        while (millis() - start < 3000) {
```
Comment (marcelvanworkum) [inline hardware/src/libs/ESP8266/ESP8266.cpp]: magic number 3000

File: hardware/src/libs/ESP8266/ESP8266.cpp (hunk 2 above)
Comment (jamiesanson) [reply hardware/src/libs/ESP8266/ESP8266.cpp]: Not actually my code. Just realised I haven't linked the library anywhere though

File: hardware/src/libs/ESP8266/ESP8266.cpp (hunk 3 above)
Comment (jamiesanson) [reply hardware/src/libs/ESP8266/ESP8266.cpp]: I think millis() makes it pretty intuitive that that means 3 seconds

File: hardware/src/libs/ESP8266/ESP8266.cpp (hunk 2 above)
Comment (jamiesanson) [reply hardware/src/libs/ESP8266/ESP8266.cpp]: @Recursively Link here https://github.com/itead/ITEADLIB_Arduino_WeeESP8266

**Comment by marcelvanworkum** [comment]
lgtm

**Pull request merged**